public/src/components/channelManagement/richTextEditor.tsx

```diff
@@ -0,0 +1,407 @@
+import React, { useCallback, useEffect, useLayoutEffect, useMemo, useState } from 'react';
+import {
+  BoldExtension,
+  ItalicExtension,
+  LinkExtension,
+  ShortcutHandlerProps,
+  TextHighlightExtension,
+  createMarkPositioner,
+} from 'remirror/extensions';
+import {
+  ComponentItem,
+  EditorComponent,
+  FloatingToolbar,
+  FloatingWrapper,
+  Remirror,
+  ToolbarItemUnion,
+  useActive,
+  useAttrs,
+  useChainedCommands,
+  useCurrentSelection,
+  useExtension,
+  useRemirror,
+  useUpdateReason,
+} from '@remirror/react';
+
+import { createStyles, WithStyles, withStyles } from '@material-ui/core';
+
+// CSS
+// Note that ReMirror/ProseMirror CSS has been added to the /src/style.css file
+const styles = createStyles({
+  fieldLabel: {
+    display: 'inline-block',
+    fontSize: '85%',
+    color: 'rgba(0 0 0 / 0.6)',
+    margin: '0 1.5em',
+  },
+  helperText: {
+    fontSize: '85%',
+    color: 'rgba(0 0 0 / 0.6)',
+    margin: '0.5em 0 0 1.5em',
+  },
+  errorText: {
+    color: 'rgba(0 0 0 / 1)',
+    backgroundColor: 'rgba(255 255 0 / 1)',
+    margin: '0.5em 0 0 1.5em',
+  },
+});
+
+// Typescript
+interface RichTextEditorProps extends WithStyles<typeof styles> {
+  disabled: boolean;
+  label?: string;
+  helperText?: string;
+  name?: string;
+  error: boolean;
+  updateCopy: (item?: string[]) => void;
+  copyData?: string | string[];
+}
+
+interface RichTextMenuProps extends WithStyles<typeof styles> {
+  disabled: boolean;
+  label: string | undefined;
+}
+
+// ReMirror/ProseMirror LINK functionality
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+function useLinkShortcut() {
+  const [linkShortcut, setLinkShortcut] = useState<ShortcutHandlerProps | undefined>();
+  const [isEditing, setIsEditing] = useState(false);
+
+  useExtension(
+    LinkExtension,
+    ({ addHandler }) =>
+      addHandler('onShortcut', props => {
```

Comment:
what is the shortcut key?

Reply:
Good question. I'll find an answer as this is the sort of thing we'd want to document in the User Guide.

Reply:
I'll play around with deleting things and see what occurs

Reply:
As far as I can tell there is no default shortcut for links. I suggest deleting `useLinkShortcut` from this file